### ISSUE TITLE
Don't escape third-party query params in requests to third-party servers

### ISSUE THREAD

**Issue by seanh** [hypothesis/via#36]
I think there's a problem with third-party query params being escaped in requests to third-party servers. I'm not sure whether this problem also exists on master or not -- if it does you can create a separate issue for it, instead of fixing it in this PR. But we will have to fix it before we can release via 3 to production.

Details:

Looking at the network requests in Chrome's dev tools I think the query params might be getting escaped in the request to the third-party for the proxied PDF file.

For example given this URL `http://localhost:9082/pdf/http://pdf995.com/samples/pdf.pdf?via.open_sidebar=1&q=1 2 3 &b=foobar` I'm seeing this request:

![Screenshot from 2020-01-16 11-20-25](https://user-images.githubusercontent.com/22498/72520856-4091eb80-3852-11ea-9d4f-40a18ce838eb.png)

Notice the `&amp;b=`: the `&` has been escaped. Of course Chrome is showing the request to our Nginx proxy, not the actual request to the third-party server itself. But I think this probably means that the query params are being escaped in the request to the third-party server as well

I had a very quick look at the code and I think the problem may be that Jinja2 is autoescaping the `pdf_url` string when [`pdfjs_viewer.html.jinja2` renders it](https://github.com/hypothesis/py-proxy/blob/d141760b88e1f06084420025811871c9ad97fa1a/py_proxy/templates/pdfjs_viewer.html.jinja2#L43). I don't know whether this is the correct solution or not but you can disable autoescaping for that string:

```diff
diff --git a/py_proxy/views.py b/py_proxy/views.py
index a9e2f53..51d2bc3 100644
--- a/py_proxy/views.py
+++ b/py_proxy/views.py
@@ -1,4 +1,5 @@
 """The views for the Pyramid app."""
+import markupsafe
 import pyramid.httpexceptions as exc
 import requests
 from pyramid import response, view
@@ -47,7 +48,7 @@ def pdf(request):
     )
 
     return {
-        "pdf_url": f"{nginx_server}/proxy/static/{pdf_url}",
+        "pdf_url": markupsafe.Markup(f"{nginx_server}/proxy/static/{pdf_url}"),
         "client_embed_url": request.registry.settings["client_embed_url"],
         "h_open_sidebar": int(request.params.get(OPEN_SIDEBAR, "0")),
         "h_request_config": request.params.get(CONFIG_FROM_FRAME, None),
 
```

With that the request looks different in Chrome:

![Screenshot from 2020-01-16 11-26-33](https://user-images.githubusercontent.com/22498/72521303-1856bc80-3853-11ea-9457-16be5156e8c2.png)

**Comment by hmstepanek** [comment]
It looks like the jinja2 template is encoding this and that `url | safe` will prevent it from doing that.

**Comment by seanh** [comment]
I think we should use `markupsafe.Markup()` in the Python (as in my diff above) because it's easier to test than templates. You can write a unit test that the returned `pdf_url` is a `Markup` object. Gives the templates less to do.

**Comment by jon-betts** [comment]
You can see this easily if you goto: http://localhost:9082/pdf/http://httpbin.org/get?a=1&b=2&c=3

The values echo'ed back are:

```
"args": {
    "a": "1", 
    "amp;b": "2", 
    "amp;c": "3"
  }, ...
```

So we are definitely messing it up.
